presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/glue/PartitionFilterBuilder.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore.glue;
```

Comment:
Actually, it seems the module dependencies may need some refactoring work. There are many presto-hive-xxxx modules and its hard to avoid circular references. Right nowI think the easiest and minimum required change is to move HiveTypeTranslator to presto-hive-common in a new folder `util`: presto-hive-common/src/main/java/com/facebook/presto/hive/util/HiveTypeTranslator.java. presto-hive and presto-hive-metastore are already dependent on presto-hive-common and thus circular dependency can be avoided.

Moreover, it seems the interface TypeTranslator is not needed. Let's remove it, and make the `translate()` method in `HiveTypeTranslator` static.

Would you please create a separate commit to do the above tasks, and put it in front of the backport commit in this PR? I'm sorry if my previous round review was not clear, but let's not duplicate the code in this file.

Reply:
HiveTypeTranslator depends on hive-apache while presto-hive-common does not. So it looks like the options are to either duplicate some code in the test file, or go back to the previous implementation (moving HiveTypeTranslator to presto-hive-metastore). If we move  HiveTypeTranslator to presto-hive-metastore, then TypeTranslator is redundant and can be removed. As it stands, TypeTranslator is used in presto-hive-metastore module and cannot be removed.

Reply:
@ClarenceThreepwood I see. Let's move it to presto-hive-metastore then.